dotnet/src/Extensions/Extensions.UnitTests/PromptTemplate/Handlebars/HandlebarsPromptTemplateTests.cs

```diff
@@ -40,67 +39,115 @@ public async Task ItRendersVariablesAsync()
         var prompt = await target.RenderAsync(this._kernel, this._arguments);
 
+        // Assert   
+        Assert.Equal("Foo Bar Baz QuxBar", prompt);
+    }
+
+    [Fact]
+    public async Task ItRendersConditionalStatementsAsync()
+    {
+        // Arrange
+        var promptConfig = InitializeHbPromptConfig("Foo {{#if bar}}{{bar}}{{else}}No Bar{{/if}}");
+        var target = (HandlebarsPromptTemplate)this._factory.Create(promptConfig);
+
+        // Act on positive
+        this._arguments["bar"] = "Bar";
+        var prompt = await target.RenderAsync(this._kernel, this._arguments);
+
         // Assert   
         Assert.Equal("Foo Bar", prompt);
+
+        // Act on negative
+        this._arguments.Remove("bar");
+        prompt = await target.RenderAsync(this._kernel, this._arguments);
+
+        // Assert   
+        Assert.Equal("Foo No Bar", prompt);
     }
 
     [Fact]
-    public async Task ItRendersAsyncFunctionsAsync()
+    public async Task ItRendersLoopsAsync()
     {
         // Arrange
-        this._kernel.ImportPluginFromObject<Foo>();
-        var template = "Foo {{Foo_Bar}} {{Foo_Baz}}";
-        var promptConfig = new PromptTemplateConfig() { TemplateFormat = HandlebarsPromptTemplateFactory.HandlebarsTemplateFormat, Template = template };
+        this._arguments["items"] = new List<string> { "item1", "item2", "item3" };
+        var promptConfig = InitializeHbPromptConfig("List: {{#each items}}{{this}}{{/each}}");
         var target = (HandlebarsPromptTemplate)this._factory.Create(promptConfig);
 
         // Act
         var prompt = await target.RenderAsync(this._kernel, this._arguments);
 
         // Assert   
-        Assert.Equal("Foo Bar Baz", prompt);
+        Assert.Equal("List: item1item2item3", prompt);
     }
 
     [Fact]
-    public async Task ItUsesDefaultValuesAsync()
+    public async Task ItRegistersCustomHelpersAsync()
     {
         // Arrange
-        var promptConfig = new PromptTemplateConfig()
+        var promptConfig = InitializeHbPromptConfig("Custom: {{customHelper}}");
+        var options = new HandlebarsPromptTemplateOptions
         {
-            TemplateFormat = HandlebarsPromptTemplateFactory.HandlebarsTemplateFormat
+            RegisterCustomHelpers = (handlebars, args) =>
+            {
+                handlebars.RegisterHelper("customHelper", (writer, context, parameters) =>
+                {
+                    writer.Write("Custom Helper Output");
+                });
+            }
         };
-        promptConfig.InputVariables.Add(new InputVariable()
-        {
-            Name = "bar",
-            Description = "Bar",
-            Default = "Bar"
-        });
-        promptConfig.InputVariables.Add(new InputVariable()
-        {
-            Name = "baz",
-            Description = "Baz",
-            Default = "Baz"
-        });
-        promptConfig.Template = "Foo {{bar}} {{baz}}";
+
+        this._factory = new(TestConsoleLogger.LoggerFactory, options);
```

Comment:
Is this going to affect other tests? The factory is used in other tests that may or may not run after this one.

Reply:
Should be a non-effect if any. This just registers the custom helper, and none of the other templates used in other tests reference the helper, so if anything, it's just an extra function that's registered in the kernel.